forum/qa-include/pages/user-profile.php

```diff
@@ -921,7 +925,10 @@
     //    Private message link
 
         $option = qa_opt('allow_private_messages');
-        if ( $option && isset($loginuserid) && ($loginuserid != $userid) && !($useraccount['flags'] & QA_USER_FLAGS_NO_MESSAGES) && !$userediting ) {
+        
+        $isBlocked = qa_db_query_sub('SELECT * FROM ^blockedPw WHERE fromUserId = # AND toUserId = # OR fromUserId = # AND fromUserId = #', $loginuserid, $useraccount['userid'], $useraccount['userid'], $loginuserid);
+
+        if ($isBlocked->num_rows === 0 && $option && isset($loginuserid) && ($loginuserid != $userid) && !($useraccount['flags'] & QA_USER_FLAGS_NO_MESSAGES) && !$userediting ) {

```

Comment:
tu powinno być yoda condition, poza tym bardzo skomplikowany ten warunek -> warto go do osobnej metody a najlepiej to w ogóle porozbijać na kilka metod chociazby same warunki

Reply:
a) Może trochę się czepiam ale nie można mówić o metodach - kod nie jest obiektowy. Co najwyżej funkcje
b) w jaki sposób mogę rozbić ten warunek? Raczej tworzenie X funkcji jest bez sensu. Mam pomysł, aby po prostu do zmiennej przypisać warunek w taki sposób
```php
$isBlockedBool = $isBlocked->num_rows === 0;
``` 
a potem w ifie
```php
if ($isBlockedBool && ...) {}
```
Mogę tak zrobić?

Reply:
Metoda i funkcja to jest to samo, są to synonimy, a podział został wprowadzony tylko i wyłącznie dla twardych głów ze szkół średnich i wyższych. Semtantycznie są to te same twory.

Reply:
Hmm, no ok nie wiedziałem gdziekolwiek nie czytałem była omówiona różnica między tymi tworami. Niemniej jednak zręcznie udało Ci się ominąć udzielenia odpowiedzi na pytanie - czy taki sposób skrócenia ifa jest ok ;).